name.abuchen.portfolio/src/name/abuchen/portfolio/online/impl/GenericJSONQuoteFeed.java

```diff
@@ -237,6 +243,16 @@ protected List<LatestSecurityPrice> parse(String url, String json, String datePa
                                 Messages.MsgErrorNumberOfDateAndCloseRecordsDoNotMatch, dates.size(), close.size())));
                 return Collections.emptyList();
             }
+            
+            BigDecimal factor;
+            if(factorString.isPresent())
+            {
+                factor = new BigDecimal(factorString.get());
```

Comment:
What if the user entered a wrong decimal format which will lead to a ``NumberFormatException``?
Maybe I missed the check. But I would suggest to check if the entered value can be parsed without error in the ``onJsonFactorChanged`` and display an indicator that the value is not correct.

Reply:
If it the user enters something invalid, the respective error message (e.g. “Not a digit“ or “Character array is missing "e" notation exponential mark“) is shown in the quote preview pane, as it is down for all error messages that might appear when parsing quotes. I think this is fine.